docs/design/devices.md

```diff
@@ -0,0 +1,123 @@
+		func ProbeGPUPlugins() []gpuTypes.GPUPlugin {
+          glog.Infof("GPUPlugin: ProbeGPUPlugins")
+          allPlugins := []gpuTypes.GPUPlugin{}
+
+          // add cuda plugin
+          cudaPlugin := cuda.ProbeGPUPlugin()
+          ...
+          // add other plugins
+          xxxxPlugin := xxxx.ProbeGPUPlugin()
+        }
+		
+	2). Discover the GPU information on host. (The host need to have CUDA installed) The way we discover GPU and its information by nvidia-smi, the same way used in NVIDIA-Docker.
+	
+	3). Collect/Update the GPU status to Scheduler, like:
+	    func IsGPUAvailable(pods []*api.Pod, gpuCapacity int) bool {
+          glog.Infof("GPUPlugin: IsGPUAvaiable()")
+          totalGPU := gpuCapacity
+          totalGPURequest := int(0)
+  
+          for _, pod := range pods {
+            totalGPURequest += getGPUResourceRequest(pod)
+          }
+  
+          glog.Infof("Kubelet: IsGPUAvailable: totalGPU: %d, totalGPURequest: %d", totalGPU, totalGPURequest)
+          return totalGPURequest == 0 || (totalGPU-totalGPURequest) >= 0
+        }
+
+	4). Allocate/Release GPU to/from container. The Plugin should could mount the GPU relative libraries to the container, like build the list of devices to be exposed in the container (e.g. /dev/nvidia0, /dev/nvidiactl, /dev/nvidia-uvm). Do it as NVIDIA-Docker. 
+
+#### 2. Container with GPU requirement.
+	1). "--device" ability to add GPU to container. The dockertools need to be extended to support the "--device", then we can assign GPU to container, also other hardware could benifit from it if we want to add more hardware such as GPU.
+	2). Allocate/Release GPU to/from container.
+	
+### Extend the Scheduler
+Scheduler has the ability to schedule and track the hardware supported in Kubelet. The algorithm need to add the ability.
+#### 1. GPU information.
+	Scheduler only can get the number of GPU information. we can call it like NVIDIA-GPU to separate it from other vendors' GPU. But in the future, we need to redesign it, like:
+	type Devices struct {
+		Name string
+		Vendor string
+		Number uint
+	}
+	then we can add more devices without change the general struct.
+	
+#### 2. The scheduler should treat GPU as dedicate resource, so can't like CPU and memory, it should as below:
+	fitsCPU := totalMilliCPU == 0 || (totalMilliCPU-milliCPURequested) >= podRequest.milliCPU
+    fitsMemory := totalMemory == 0 || (totalMemory-memoryRequested) >= podRequest.memory
+    fitGPU := (totalGPU - GPURequested) >= podRequest.GPU
+
+
+### Extend the resource model
+
+Resource quantities
+Should include number to describe how many GPUs on a machine.
+
+
+### yaml file example:
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nvidia-cuda
+spec:
+  containers:
+  - name: nvidia-cuda
+    image: nvidia_cuda:v1
+    command: [ "sh", "/test.sh"]
+    resources:
+      requests:
+        cpu: 100m
+        memory: 100Mi
+        nvidiagpu: 1
```

Comment:
For now.  Makes it clear feature is alpha.  Later, maybe we rename it to `alpha.kubernetes.io/nvidia-gpu`, or maybe we find a way to abstract GPU vendors and can call it "gpu"